Automattic/woocommerce-payments
# Subscriptions compatibility for order tracking updates

Fixes 585-gh-Automattic/woocommerce-payments-server 

Relies on #1295.

#### Changes proposed in this Pull Request

Adds Subscription Compatibility for Order Update Events

Some order updates were not getting tracked because the child-orders which are created for a Subscription do not have an `intent_id` attached. This adds logic to the Subscriptions compatibility WCPay gateway to get the details of the parent order when a subscription is updated.

#### Testing instructions

- Checkout this branch and server branch 590-gh-Automattic/woocommerce-payments-server (if not already merged).
- Make sure the async jobs watcher is running on your sandbox and that your client is pointed at your sandbox.
- Create a subscription order by purchasing it via the checkout. 
- Check **Tools -> ActionScheduler** and verify that `wcpay_create_order` and `wcpay_update_order` events are present for the newly created subscription order.
- Run these jobs and verify that the events appear in the async jobs watcher and in the [Sift Console](https://console.sift.com/developer/api-logs?abuse_type=account_abuse).
- Go to the account of the user who purchased the order (**My Account -> Subscriptions**) and make a change to the subscription (for example, cancelling it or changing the payment method)
- Verify that this created an update order event in the **Tools -> ActionScheduler** section (this should be visible in the 'Pending' jobs section). Run it and verify it appears in the async jobs watcher and in Sift console.

Note that there is still an issue when it comes to free trial subscriptions. Once the free trial is over, the renewal order event should send an event, however the free trial order creation currently does not send an event because there is no Payment Method attached to the order, which is a requirement for tracking. This PR however stops any errors/alerts from occurring because of this (previously, we were seeing some errors in Slack related to subscriptions).



-------------------
